.github/workflows/integration.yml

```
@@ -250,24 +255,38 @@ jobs:
           sudo rm -rf "$AGENT_TOOLSDIRECTORY"
           echo "=== After cleanup:"
           df -h
+
       - uses: actions/checkout@v4
+        with:
+          path: ./agoric-sdk
+
       - name: docker build (sdk)
         # Produces ghcr.io/agoric/agoric-sdk:unreleased used in the following upgrade test.
         # run: cd packages/deployment && ./scripts/test-docker-build.sh | $TEST_COLLECT
         # XXX skip TAP test output and collection for now; it hides the output from the logs
         run: make docker-build-sdk
-        working-directory: packages/deployment
-      - uses: ./.github/actions/restore-node
+        working-directory: agoric-sdk/packages/deployment
+
+      - id: restore-golang
+        uses: ./agoric-sdk/.github/actions/restore-golang
+        with:
+          go-version: '${{ env.GOLANG_VERSION }}'
+          path: ./agoric-sdk
+
+      - id: restore-node
+        uses: ./agoric-sdk/.github/actions/restore-node
         with:
-          node-version: '18.x'
           # Rebuilding the SDK image with resolved endo packages is not currently supported
           # and since we can't build core eval submissions from the SDK using a newer endo,
           # simply ignore any endo branch integration (this means we don't have full coverage)
           ignore-endo-branch: 'true'
-        id: restore-node
+          node-version: '18.x'
+          path: ./agoric-sdk
+
```

Let's set the `xsnap-random-init` option even if we don't remove the loadgen from the deployment-test yet.

```suggestion
          # Forces xsnap to initialize all memory to random data, which increases
          # the chances the content of snapshots may deviate between validators
          xsnap-random-init: '1'

```

---

Done